src/components/visualization/vegaLiteViz/vegaLiteViz.tsx

```diff
@@ -48,21 +52,62 @@ function VegaLiteViz(props: VegaLiteData) {
     id: 'rustic-vega-lite-tooltip',
   }
 
+  const menuItems: PopoverMenuItem[] = [
+    {
+      label: 'Save as SVG',
+    },
+    {
+      label: 'Save as PNG',
+    },
+  ]
+
+  function getScaleFactor(
+    format: 'svg' | 'png',
+    scaleFactor: number | { svg?: number; png?: number } | undefined
+  ) {
+    if (scaleFactor) {
+      if (typeof scaleFactor === 'number') {
+        return scaleFactor
+      } else {
+        return scaleFactor[format]
+      }
+    }
+  }
+
   function renderChart() {
     if (chartRef.current && props.spec) {
       const options = {
         config: { font: defaultFont },
         ...props.options,
         theme: isDarkTheme ? props.theme?.dark : props.theme?.light,
         tooltip: tooltipOptions,
+        actions: false,
       }
 
       if (!props.options?.config?.font) {
         options.config.font = defaultFont
       }
 
       VegaEmbed(chartRef.current, props.spec, options)
-        .then(() => {
+        .then((result) => {
+          const scaleFactor = result.embedOptions.scaleFactor
+          typeof scaleFactor === 'number' ? scaleFactor : scaleFactor?.svg
+          const fileName =
+            result.embedOptions.downloadFileName || 'visualization'
+          const formats = ['svg', 'png']
+
+          formats.map((format, index) => {
+            result.view
+              .toImageURL(
+                format,
+                getScaleFactor(format as 'svg' | 'png', scaleFactor)
+              )
```

Comment:
I dont think we need `getScaleFactor`
```suggestion
            const opts = result.embedOptions
            const scaleFactor = isObject(opts.scaleFactor) ? opts.scaleFactor[format] : opts.scaleFactor;
            result.view
              .toImageURL(
                format,
                scaleFactor
              )
```

Reply:
Javascript doesn't have `isObject`. I used `typeof opts.scaleFactor === 'object'` instead.

Reply:
yes that was just to illustrate what we could do